cpp/include/cuspatial/detail/distance/linestring_distance.cuh

```diff
@@ -48,25 +50,43 @@ OutputIt pairwise_linestring_distance(MultiLinestringRange1 multilinestrings1,
                         typename MultiLinestringRange2::point_t>(),
                 "All input types must be cuspatial::vec_2d with the same value type");
 
-  CUSPATIAL_EXPECTS(multilinestrings1.size() == multilinestrings2.size(),
-                    "Inputs must have the same number of rows.");
+  CUSPATIAL_EXPECTS(lhs.size() == rhs.size(), "Inputs must have the same number of rows.");
 
-  if (multilinestrings1.size() == 0) return distances_first;
+  if (lhs.size() == 0) return distances_first;
 
+  // Make views to the segments in the multilinestring
+  auto lhs_segments       = lhs._segments(stream);
+  auto lhs_segments_range = lhs_segments.view();
+
+  // Make views to the segments in the multilinestring
+  auto rhs_segments       = rhs._segments(stream);
+  auto rhs_segments_range = rhs_segments.view();
+
+  auto thread_bounds =
+    detail::compute_segment_thread_bounds(lhs_segments_range.multigeometry_count_begin(),
+                                          lhs_segments_range.multigeometry_count_end(),
+                                          rhs_segments_range.multigeometry_count_begin(),
+                                          stream);
+  // Initialize the output range
   thrust::fill(rmm::exec_policy(stream),
                distances_first,
-               distances_first + multilinestrings1.size(),
+               distances_first + lhs.size(),
                std::numeric_limits<T>::max());
 
   std::size_t constexpr threads_per_block = 256;
-  std::size_t const num_blocks =
-    (multilinestrings1.num_points() + threads_per_block - 1) / threads_per_block;
+  std::size_t num_threads                 = 1e8;
```

Comment:
Rather than hard coding, this may be a job for the occupancy API...
https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__OCCUPANCY.html